.github/workflows/unit-test-backend.yaml

```diff
@@ -0,0 +1,51 @@
+###############################################################
+# Copyright (c) 2024 Contributors to the Eclipse Foundation
+# Copyright (c) 2024 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e.V.
+# (represented by Fraunhofer ISST)
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+###############################################################
+---
+
+name: Unit Test - Backend
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  unit-test-backend:
+    if: github.repository_owner == 'eclipse-tractusx' # prevent running on forks
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

Comment:
It's probably better to use a release version, i.e.   

```uses: actions/checkout@releases/v3```

Reply:
The commit hash points to the commit SHA of a specific release version. This is done to security reasons, as tags may be changed but commits not. Dependapot also understands that as you can see [here](https://github.com/eclipse-tractusx/puris/pull/627/files).

I'm with you, that it would reduce maintenance efforts to use the major release only, but would keep it due to the security best practice (yes, very small benefit).

Reply:
Please do as suggested or explain, why you disagree